### PR TITLE
add setup steps in tutorial

### PR DESCRIPTION
**Pull request by Demezy**
Fixes  source of #1011 and similar https://github.com/fzyzcjy/flutter_rust_bridge/issues?q=%27stdbool.h%27+file+not+found

Have **not** changed source code, only add info in docs. So, skip code test part 

## Checklist

- [x] An issue to be fixed by this PR is listed above.
- [ ] New tests are added to ensure new features are working. End-to-end tests are usually in the `./frb_example/pure_dart` example, more specifically, `rust/src/api.rs` and `dart/lib/main.dart`.
- [x] The code generator is run and the code is formatted (e.g. via `just refresh_all`).
- [x] If this PR adds/changes features, documentations (in the `./book` folder) are updated.
- [ ] CI is passing.

### CODE REVIEW

**Comment by Demezy** [comment]
I put this to [tutorial](https://cjycode.com/flutter_rust_bridge/tutorial_with_flutter.html) part because it is the most explicit location for user that comes for tutorial :) 
Another variant is to add to [dependencies part](https://cjycode.com/flutter_rust_bridge/integrate/deps.html) and just mention it

File: book/src/tutorial_with_flutter.md [hunk 1]
```diff
@@ -24,6 +24,21 @@ Please [install Flutter](https://flutter.dev/docs/get-started/install) (optional
 git clone https://github.com/fzyzcjy/flutter_rust_bridge && cd flutter_rust_bridge/frb_example/with_flutter
 ```
 
+## Setup dependencies
+
+For non-debian based distributions a few tricks required.
```
Comment (fzyzcjy) [inline book/src/tutorial_with_flutter.md:29]:
```suggestion
For non-debian based distributions, there are a few prerequisites:
```

File: book/src/tutorial_with_flutter.md [hunk 2]
```diff
@@ -24,6 +24,21 @@ Please [install Flutter](https://flutter.dev/docs/get-started/install) (optional
...
+
+- clang
+- llvm-libs
+- glibc
+
+Set environment variable in your shell profile (`.bashrc`, `.zshrc`, etc).
+
+```bash
+export CPATH="$(clang -v 2>&1 | grep "Selected GCC installation" | rev | cut -d' ' -f1 | rev)/include"
+```
```
Comment (fzyzcjy) [inline book/src/tutorial_with_flutter.md:40]: So, you mentioned "non-debian" users. This includes:

- [ ] windows
- [ ] linux (except debian)
- [ ] macos (M1, Intel)

Is it true that all these kinds of users need the setup? (IMHO I myself do not need it; I am on macos intel)

**Review by fzyzcjy — COMMENTED** [review]
Good job!

**Comment by fzyzcjy** [comment]
> Another variant is to add to [dependencies part](https://cjycode.com/flutter_rust_bridge/integrate/deps.html) and just mention it

Personally I prefer this, since these are also dependencies. If we mention them in the with_flutter tutorial using a section, the readers may get distracted.

File: book/src/tutorial_with_flutter.md (hunk 2 above)
Comment (Demezy) [reply book/src/tutorial_with_flutter.md:40]: You are right, I mean Linux other than debian. At least during issues research I have not met others

File: book/src/integrate/deps.md [hunk 3]
```diff
@@ -65,3 +65,20 @@ Add these lines to `Cargo.toml`:
...
+## System dependencies
+
+For non-debian based Linux distributions, there are a few prerequisites:
+Ensure that packages are up to date (or install by demand).
+
+- clang
+- llvm-libs
+- glibc
+
+> restart may be required
```
Comment (Demezy) [inline book/src/integrate/deps.md:77]: I moved parts, so now it is more precise.
How to better style this part?


Please check when you'll have time

Comment (fzyzcjy) [reply book/src/integrate/deps.md:77]: That LGTM, I will restyle after merge. Good job!

Comment (Demezy) [reply book/src/integrate/deps.md:77]: Ok! Thank you!

**Comment by Demezy** [comment]
I resolved threads, so now PR is ready to merge

**Comment by fzyzcjy** [comment]
@all-contributors please add @Demezy for doc

**Pull request merged**